### ISSUE TITLE
Conditionally Compile oos_util_lob.write_file

### ISSUE THREAD

**Issue by martindsouza** [OraOpenSource/oos-utils#156]
From #56 

`oos_util_lob.write_file` requires access to `utl_file`. Not all users may have access to this package. For that reason I've removed it for 1.0.0. Will look at conditional compilation for 1.1.0

@janihur FYI

**Comment by zhudock** [comment]
Not sure if there's a better way, but in a work project we've recently been using 

`select 1 from user_tab_privs where grantee = user and table_name = 'UTL_FILE' and owner = 'SYS'`

**Comment by martindsouza** [comment]
@zhudock thanks for the query. What about roles that a user may have access to? Will that be included in `user_tab_privs`?

**Comment by zhudock** [comment]
@martindsouza good call. I forgot about roles.  It looks like role_tab_privs does NOT include the TYPE column.

`select 1 from role_tab_privs where role in (select granted_role from user_role_privs where username = user) and table_name = 'UTL_FILE' and owner = 'SYS';`

**Comment by martindsouza** [comment]
@zhudock and @janihur  can you please review what I did for this ticket? I'm still on the fence as to "hiding" the procedures entirely when user doesn't have access to `sys.utl_file` or moving the conditional compilation (CC) code into each procedure and raising an error message if they don't have access. Both have pros and cons.

**Comment by zhudock** [comment]
I'd vote for the CC code inside each proc and raising an exception if they don't have access to `sys.utl_file`. That way instead of the procedures being "hidden" the developer, or DBA, can be made aware that some additional functionality exists but they're missing the appropriate grant to use it.

**Comment by martindsouza** [comment]
@zhudock good feedback. I've updated `oos_util_lob` accordingly. Only difference from this logic will be `oos_util_apex` as all the parameters reference APEX types.
